planner/core/rule_partition_processor.go

```diff
@@ -624,3 +651,139 @@ func (s *partitionProcessor) makeUnionAllChildren(ds *DataSource, pi *model.Part
 	unionAll.SetSchema(ds.schema.Clone())
 	return unionAll, nil
 }
+
+func (s *partitionProcessor) pruneRangeColumnsPartition(ds *DataSource, pi *model.PartitionInfo) (LogicalPlan, error) {
+	result := fullRange(len(pi.Definitions))
+
+	if len(pi.Columns) != 1 {
+		// We only support single column.
+		return s.makeUnionAllChildren(ds, pi, result)
+	}
+
+	prunner, err := makePartitionRangePrunner2(ds, pi)
+	if err == nil {
+		result = partitionRangeForCNFExpr(ds.ctx, ds.allConds, prunner, result)
+	}
+	return s.makeUnionAllChildren(ds, pi, result)
+}
+
+var _ partitionRangePrunner = &partitionRangePrunner2{}
+
+// partitionRangePrunner2 is used by 'partition by range columns'.
```

Comment:
+1 maybe  rename just as comments? 1, 2 is hard to understand at first glance
`partitionRangePrunner1` => `partitionRangePrunner`
`partitionRangePrunner2` => `partitionRangeColumnsPrunner`

Reply:
Seems a good idea